pkg/validator/validators.go

```diff
@@ -59,6 +60,20 @@ func FakeValidator(req *admissionv1.AdmissionRequest) (*admissionv1.AdmissionRes
 */
 type validateFunc func(req *admissionv1.AdmissionRequest) (*admissionv1.AdmissionResponse, error)
 
+func trafficTargetValidator(req *admissionv1.AdmissionRequest) (*admissionv1.AdmissionResponse, error) {
+	trafficTarget := &smiAccess.TrafficTarget{}
+	if err := json.NewDecoder(bytes.NewBuffer(req.Object.Raw)).Decode(trafficTarget); err != nil {
+		return nil, err
+	}
+
+	if trafficTarget.Spec.Destination.Namespace != trafficTarget.Namespace {
+		return nil, errors.Errorf("The traffic target namespace (%s) must match spec.Destination.Namespace (%s)",
+			trafficTarget.Namespace, trafficTarget.Spec.Destination.Namespace)
```

Comment:
I'm not opposed returning `errors.Errorf`, but that would make it hard in the future should we want to differentiate between different types of errors returned by this validator.

I don't know that we have settled on a standard for this repo.
We tend to create an error and place it in the `errors.go` file in the package.
```go
var ErrTrafficTargetNamespaceMismatch = errors.New("traffic target namespace mismatch")
```
and then here `return nil, ErrTrafficTargetNamespaceMismatch`
but also before the return log the full statement
```go
log.Error().Err(ErrTrafficTargetNamespaceMismatch).Msgf("The traffic target namespace (%s) must match spec.Destination.Namespace (%s)", trafficTarget.Namespace, trafficTarget.Spec.Destination.Namespace)`
```

Reply:
@draychev the recommended way to log errors currently is as follows:
1. For dynamic error messages without error codes, use `errors.Errorf(), errors.New()`.
2. For static errors messages previously defined in errors.go, define it in `pkg/errcode/errcode.go` instead so it can be logged similar to other error codes with the ability to publish metrics per error code.

I am not opposed to defining errors in errors.go if they don't need additional data to be encoded in them, though the errcode model should be adopted to get the benefits of tooling we build to inspect error codes instead of error messages.